ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -941,6 +955,10 @@ data NodeEvents blk ev = NodeEvents
     -- ^ the point of every 'ChainDB.InvalidBlock' event
   , nodeEventsTipBlockNos :: ev (SlotNo, WithOrigin BlockNo)
     -- ^ 'ChainDB.getTipBlockNo' for each node at the onset of each slot
+  , nodeEventsTxRequests  :: ev ([GenTxId blk], RecentTxIds (GenTxId blk))
+    -- ^ every request for transactions made by the node-to-node
+    -- 'TxSubmission' mini-protocol along with the 'RecentTxIds' at the time
+    -- each request was made.
```

Comment:
So we store the `RecentTxIds` each time a request for txs is sent. I suppose it will be fine, but can you check that the memory usage while running the tests doesn't increase by too much? And can you check what the impact on the run time is? (Use the same seed to run the tests a few times with and without the new checks.) The tests are already slow enough :slightly_smiling_face: